validator.js

```diff
@@ -3,6 +3,31 @@ var fs = require('fs');
 var path = require('path');
 var resumeJson = require('./resume');
 
+
+/* Register custom validators */
+
+/**
+ * Register a custom date validator "resume-date" that allows:
+ *
+ * - "YYYY-MM-DD" : year, month, and day
+ * - "YYYY-MM"    : year and month
+ * - "YYYY"       : year
+ * - ""           : empty string represents the present
+ */
+ZSchema.registerFormat('resume-date', function (date) {
+  if (typeof date !== 'string') {
```

Comment:
If it is not a string why should it return `true` ? If you wish to check for a date object you can replace the `if` condition with

```js
if (date instanceof Date)